### PR TITLE
Expose frontend in HTTPS

### LINKED ISSUE #14136: [Feature request]: Option to expose frontend throught HTTPS
### Is your feature request related to a problem? Please describe

It would be great to make the frontend exposed through HTTPS.

### Describe the solution you'd like

in configuration.yaml, add :
```yaml
frontend:
  secure:
    ssl_cert: [path to cert file]
    ssl_key: [path to key file]
```

then adapt code in the `start()` function in [lib/extension/frontend.ts](https://github.com/Koenkk/zigbee2mqtt/blob/master/lib/extension/frontend.ts) 

### Describe alternatives you've considered

It can be handled with a proxy like Traefik... I do use one for my websites exposed on the internet.
but I do not want to use such proxy for LAN access.

### Additional context

I may try to code this by myself as it is not really complicated (I suppose), and submit a pull request.

### PR DESCRIPTION
**Pull request by PheelTi**
PR for HTTPS exposure, see feature request #14136 
patch to expose frontend in HTTPS.
the chose implementation for simplicity is the add of 2 configuration params under 'frontend' : 
- 'ssl_cert' path to SSL certificate
- 'ssl_key' path to SSL key

100% test coverage.

Closes #14136

### CODE REVIEW

File: lib/extension/frontend.ts [hunk 1]
```diff
@@ -19,6 +21,8 @@ export default class Frontend extends Extension {
     private mqttBaseTopic = settings.get().mqtt.base_topic;
     private host = settings.get().frontend.host;
     private port = settings.get().frontend.port;
+    private ssl_cert = settings.get().frontend.ssl_cert;
+    private ssl_key = settings.get().frontend.ssl_key;
```
Comment (Koenkk) [inline lib/extension/frontend.ts:25]: Can you make this camelcase (`sslKey`/`sslCert`)

File: lib/extension/frontend.ts [hunk 2]
```diff
@@ -33,8 +37,29 @@ export default class Frontend extends Extension {
         this.eventBus.onMQTTMessagePublished(this, this.onMQTTPublishMessage);
     }
 
+    private isHttpsConfigured():boolean {
+        if (this.ssl_cert && this.ssl_key &&
+            this.ssl_cert.length > 0 && this.ssl_key.length > 0) {
+            if (!fs.existsSync(this.ssl_cert) || !fs.existsSync(this.ssl_key)) {
+                logger.warn(`defined ssl_cert or ssl_key file path does not exists : '${this.ssl_cert}', server won't be secured.`); /* eslint-disable-line max-len */
```
Comment (Koenkk) [inline lib/extension/frontend.ts:44]: I think an `logger.error` suits better here

File: lib/extension/frontend.ts [hunk 3]
```diff
@@ -33,8 +37,29 @@ export default class Frontend extends Extension {
         this.eventBus.onMQTTMessagePublished(this, this.onMQTTPublishMessage);
     }
 
+    private isHttpsConfigured():boolean {
+        if (this.ssl_cert && this.ssl_key &&
+            this.ssl_cert.length > 0 && this.ssl_key.length > 0) {
```
Comment (Koenkk) [inline lib/extension/frontend.ts:42]: Empty string evaluate to false so no need to check the length:

Screenshot 2022-12-03 at 10 40 31

**Comment by Koenkk** [comment]
Looks good, I added a few minor comments.

Could you also make a PR for the docs? https://github.com/Koenkk/zigbee2mqtt.io/edit/master/docs/guide/configuration/frontend.md

File: lib/extension/frontend.ts (hunk 2 above)
Comment (PheelTi) [reply lib/extension/frontend.ts]: done, I also changed the error message : 
`defined ssl_cert '${this.sslCert}' or ssl_key '${this.sslKey}' file path does not exists, server won't be secured.`

File: lib/extension/frontend.ts (hunk 1 above)
Comment (PheelTi) [reply lib/extension/frontend.ts]: done.

File: lib/extension/frontend.ts (hunk 3 above)
Comment (PheelTi) [reply lib/extension/frontend.ts]: yesn done... I am more a Java deceloper sorry 🙂

**Comment by PheelTi** [comment]
Thank you !
remarks taken in account, and PR for docs submitted : https://github.com/Koenkk/zigbee2mqtt.io/pull/1738

**Comment by PheelTi** [comment]
Closes #14136

**Comment by Koenkk** [comment]
Big thanks!

**Pull request merged**